modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxCleanupRequestSender.java

```diff
@@ -74,27 +76,42 @@ public CompletableFuture<Void> cleanup(String node, UUID txId) {
         return sendCleanupMessageWithRetries(false, null, txId, node, null);
     }
 
+    private CompletableFuture<Void> cleanup(
+            Collection<TablePartitionId> partitionIds,
+            Map<TablePartitionId, String> originalPartitions,
+            boolean commit,
+            @Nullable HybridTimestamp commitTimestamp,
+            UUID txId
+    ) {
+        return placementDriverHelper.findPrimaryReplicas(partitionIds)
```

Comment:
Here I am doubting whether seeking out primary is necessary because we are already in the commit phase and we know nodes where all resources are open.